httpx/_dispatch/asgi.py

```diff
@@ -1,9 +1,28 @@
+import typing
 from typing import Callable, Dict, List, Optional, Tuple
 
 import httpcore
+import sniffio
 
 from .._content_streams import ByteStream
 
+if typing.TYPE_CHECKING:  # pragma: no cover
+    import asyncio
+    import trio
```

Comment:
Okay, so `trio` might not be installed.

Perhaps we should be doing something like this?...

```python
try:
    import trio
    Event = typing.Union[asyncio.Event, trio.Event]
except ImportError:
    Event = asyncio.Event
```

Also it's not obvious to me that we need to bother with the `if typing.TYPE_CHECKING:` dance? We *usually* only use that if it's required to avoid otherwise circular imports.

Reply:
I tried what you're suggesting but unfortunately run into an error like:
```
error: Cannot assign multiple types to name "Event" without an explicit "Type[...]" annotation
```

Seems like type aliases need to be statically resolvable: https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases

Reply:
Okay. We can't rely on the `trio` package necessarily being installed tho.
(Even if we're limiting this to `if typing.TYPE_CHECKING:`)

One other option might be to just under-specify the type in this case, as a lazy `typing.Any`. We're not exposing the type info or using that function outside of this scope, so perhaps that's a sensibly practical option?

Reply:
The `if typing.TYPE_CHECKING` does actually help here. If `trio` (actually, `trio-typing`) is not installed, mypy is still happy because we use `--ignore-missing-imports` so it just ignores `import trio` and `trio.Event()`.

Reply:
@tomchristie still want me to make that a `typing.Any`?